File: src/main/java/uk/co/liamjdavison/annotations/SparkController.java
package uk.co.liamjdavison.annotations;

import java.lang.annotation.*;

/**
 * Classes annotated with @SparkController will be initiated with the server is started, so that all their corresponding routes will exist
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE) //on class level
@Documented
public @interface SparkController {
}

File: src/main/kotlin/uk/co/liamjdavison/kotlinsparkroutes/annotations/SparkController.kt
package uk.co.liamjdavison.kotlinsparkroutes.annotations

/**
 * Classes annotated with @SparkController will be initiated with the server is started, so that all their corresponding routes will exist
 */
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS)
annotation class SparkController